python/pyarrow/tests/test_csv.py

```diff
@@ -54,18 +53,32 @@ def generate_col_names():
             yield first + second
 
 
+def split_rows(arr, num_cols, num_rows):
+    # Split a num_cols x num_rows array into rows
+    for i in range(0, num_rows*num_cols, num_cols):
+        yield list(itertools.islice(arr, i, i + num_cols))
```

Comment:
I've just noticed that you're going through `itertools.islice` just to accumulate the results into a list, so you can probably slice the input directly, which is going to be much more efficient:
```suggestion
    for i in range(0, num_rows * num_cols, num_cols):
        yield arr[i:i + num_cols]
```
```